lib/rucio/core/rule.py

```diff
@@ -888,15 +888,16 @@ def list_associated_rules_for_file(scope, name, *, session: "Session"):
     :param session: The database session in use.
     :raises:        RucioException
     """
-
     rucio.core.did.get_did(scope=scope, name=name, session=session)  # Check if the did acually exists
-    query = session.query(models.ReplicationRule).\
+    query = session.query(models.ReplicationRule, models.DataIdentifier.bytes).\
         with_hint(models.ReplicaLock, "INDEX(LOCKS LOCKS_PK)", 'oracle').\
         join(models.ReplicaLock, models.ReplicationRule.id == models.ReplicaLock.rule_id).\
         filter(models.ReplicaLock.scope == scope, models.ReplicaLock.name == name).distinct()
```

Comment:
Doesn’t this use a Cartesian product? What is missing is another `JOIN` clause.

```python
join(models.DataIdentifier, and_(models.ReplicationRule.scope == models.DataIdentifier.scope, models.ReplicationRule.name == models.DataIdentifier.name)).\
```

Reply:
Yep, you're correct. Forgive my 3-month-ago self's naivety :]

Reply:
It would appear that this was not carried over in the latest iteration of your PR.

Reply:
Ack, the rebase got me. Yeah correcting along with investigating those failing tests.